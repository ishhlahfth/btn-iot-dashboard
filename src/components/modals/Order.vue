<template>
  <div
    class="grid grid-flow-row sm:grid-flow-col gap-6 merchant-modal-content inner-modal-fixed modal-xl overflow-auto"
  >
    <div class="grid md:grid-flow-col gap-8 auto-rows-max overflow-auto">
      <div class="grid gap-4 md:gap-8">
        <div class="divide-y divide-grey-4 font-medium">
          <p class="font-medium pb-4 md:text-base">Order</p>
          <div class="pt-4 grid grid-cols-2 md:grid-cols-4 gap-4">
            <div>
              <p class="text-grey-2">PO Number</p>
              <p>O1090F</p>
            </div>
            <div>
              <p class="text-grey-2">Order Date</p>
              <p>23 Apr 2021 11:20</p>
            </div>
            <div>
              <p class="text-grey-2">Last Updated</p>
              <p>23 Apr 2021 11:20</p>
            </div>
            <div>
              <p class="text-grey-2">Status</p>
              <p class="text-royal">Pending</p>
            </div>
          </div>
        </div>
        <div class="divide-y divide-grey-4 font-medium">
          <p class="font-medium pb-4 md:text-base">Delivery</p>
          <div class="pt-4 grid grid-cols-2 md:grid-cols-4 gap-4">
            <div>
              <p class="text-grey-2">PO Number</p>
              <p>O1090F</p>
            </div>
            <div>
              <p class="text-grey-2">Order Date</p>
              <p>23 Apr 2021 11:20</p>
            </div>
            <div>
              <p class="text-grey-2">Last Updated</p>
              <p>23 Apr 2021 11:20</p>
            </div>
            <div>
              <p class="text-grey-2">Status</p>
              <p class="text-royal">Pending</p>
            </div>
          </div>
        </div>
        <div class="divide-y divide-grey-4 font-medium">
          <p class="font-medium pb-4 md:text-base">Seller</p>
          <div class="pt-4 grid grid-cols-2 md:grid-cols-4 gap-4">
            <div>
              <p class="text-grey-2">Name</p>
              <p>Herba Kitchen</p>
            </div>
            <div>
              <p class="text-grey-2">Location</p>
              <p>Jakarta Selatan</p>
            </div>
            <div>
              <p class="text-grey-2">Phone Number</p>
              <p>0896-3340-1529</p>
            </div>
          </div>
        </div>
        <div class="divide-y divide-grey-4 font-medium">
          <p class="font-medium pb-4 md:text-base">Buyer</p>
          <div class="pt-4 grid grid-cols-2 md:grid-cols-4 gap-4">
            <div>
              <p class="text-grey-2">Name</p>
              <p>Kania</p>
            </div>
            <div>
              <p class="text-grey-2">Location</p>
              <p>Jakarta Barat</p>
            </div>
            <div>
              <p class="text-grey-2">Phone Number</p>
              <p>0896-3340-1529</p>
            </div>
          </div>
        </div>
        <div class="divide-y divide-grey-4 font-medium">
          <p class="font-medium pb-4 md:text-base">Product</p>
          <div class="pt-4 grid grid-cols-2 md:grid-cols-4 gap-4">
            <div>
              <p class="text-grey-2">Name</p>
              <p>Kania</p>
            </div>
            <div>
              <p class="text-grey-2">Price Per Product</p>
              <p>Jakarta Barat</p>
            </div>
            <div>
              <p class="text-grey-2">Category</p>
              <p>0896-3340-1529</p>
            </div>
            <div>
              <p class="text-grey-2">Phone Number</p>
              <p>0896-3340-1529</p>
            </div>
          </div>
        </div>
      </div>
      <div class="divide-y divide-grey-4 font-medium">
        <p class="font-medium pb-4 md:text-base">Payment</p>
        <div class="pt-4 grid grid-cols-2 md:grid-cols-none gap-4 md:gap-8">
          <div>
            <p class="text-grey-2">Delivery Fee</p>
            <p>Rp 8.000</p>
          </div>
          <div>
            <p class="text-grey-2">Item Price</p>
            <p>Rp 31.000</p>
          </div>
          <div>
            <p class="text-grey-2">Commission</p>
            <p>Rp 2.000</p>
          </div>
          <div>
            <p class="text-grey-2">Total Price</p>
            <p>Rp 49.000</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'OrderDetail',
};
</script>

<style></style>
