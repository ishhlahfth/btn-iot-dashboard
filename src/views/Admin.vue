<template>
  <help-modal v-model="addModal">
    <admin-add-edit
      @close="handleClose"
      @reload="generateFetchData"
      adminType="add"
      :roles="selectRoles"
    />
  </help-modal>
  <help-modal v-model="editModal">
    <admin-add-edit
      @close="handleClose"
      @reload="generateFetchData"
      adminType="edit"
      :roles="selectRoles"
    />
  </help-modal>
  <help-modal v-model="detailModal">
    <admin-detail />
  </help-modal>
  <help-modal v-model="deleteConfirmation">
    <confirmation
      title="Delete confirmation"
      message="This action cannot be undone. Are you sure you want to delete this user permanently?"
      :confirm-loading="deleteLoading"
      loading-label="deleting"
      @close="deleteConfirmation = false"
      @cancel="deleteConfirmation = false"
      @confirm="deleteUser"
    />
  </help-modal>
  <div class="p-4 sm:p-6 grid gap-4 sm:gap-6">
    <div class="w-full flex justify-between">
      <p class="text-heading2 font-semibold">Admin</p>
      <help-button label="add" icon="plus" @click="addModal = true" v-if="adminAccess.create" />
    </div>
    <div>
      <form @submit.prevent="getAdmins">
        <help-input v-model="searchValue" placeholder="Search admin name here" search-bar />
      </form>
    </div>
    <div class="overflow-hidden">
      <help-table
        path="employees"
        :columns="columns"
        :loading="loading"
        :rows="admins"
        :pagination="adminPagination"
        @onChangePagination="getAdmins($event)"
        @sort="getAdmins($event)"
      >
        <template v-slot:body="{ data, column, row }">
          <p
            v-if="column === 'detail'"
            class="text-royal font-medium cursor-pointer"
            @click="handleDetail(row, 'detail')"
          >
            See Detail
          </p>
          <p v-if="column === 'address'">{{ data ? truncate(data) : '' }}</p>
          <p v-if="column === 'phone_number'">{{ generatePhoneNumber(data) }}</p>
          <p v-if="column === 'role'">{{ generateRoleName(row) }}</p>
          <div v-if="column === 'is_active'">
            <help-toggle
            :modelValue="is_active[`${row.id}`]"
            @change="handleActiveAdmin(row)"
            />
          </div>
          <div v-if="column === 'actions'" class="grid grid-flow-col gap-1 auto-cols-max">
            <help-button icon-only icon="edit" @click="handleDetail(row, 'edit')" v-if="adminAccess.update" />
            <help-button
            v-if="adminAccess.delete"
            bg-color="flame"
            color="white"
            icon="trash"
            icon-only
            @click="openConfirmation(row.id)" />
          </div>
        </template>
      </help-table>
    </div>
  </div>
</template>

<script>
import HelpButton from '@/components/atoms/Button.vue';
import HelpInput from '@/components/atoms/Input.vue';
import HelpModal from '@/components/templates/Modal.vue';
import HelpTable from '@/components/templates/Table.vue';
import AdminAddEdit from '@/components/modals/AdminAddEdit.vue';
import AdminDetail from '@/components/modals/AdminDetail.vue';
import HelpToggle from '@/components/atoms/Toggle.vue';
import Confirmation from '@/components/modals/Confirmation.vue';
import { useToast } from 'vue-toastification';
import mixin from '@/mixin';
import API from '@/apis';

export default {
  name: 'Admin',
  mixins: [mixin],
  components: {
    HelpButton,
    HelpInput,
    HelpModal,
    HelpTable,
    AdminAddEdit,
    HelpToggle,
    AdminDetail,
    Confirmation,
  },
  setup() {
    const toast = useToast();
    return { toast };
  },
  data() {
    return {
      searchValue: '',
      columns: [
        { field: 'name', label: 'name' },
        { field: 'email', label: 'email' },
        { field: 'address', label: 'address' },
        { field: 'phone_number', label: 'phone number' },
        { field: 'role', label: 'role' },
        { field: 'detail', label: 'detail' },
        { field: 'is_active', label: 'status', align: 'center' },
        { field: 'actions', label: '', align: 'center' },
      ],
      admins: [],
      roles: [],
      selectRoles: [],
      admin: {},
      adminPagination: {
        limit: 10,
        offset: 0,
      },
      loading: false,
      detailModal: false,
      is_active: {},
      addModal: false,
      editModal: false,
      adminAccess: {
        create: false,
        update: false,
        delete: false,
      },
      deleteConfirmation: false,
      deleteLoading: false,
    };
  },
  methods: {
    async getRoles() {
      try {
        const {
          data: { data },
        } = await API.get('/roles?offset=0&limit=100&group=INTERNAL_DASHBOARD');
        this.roles = data.map((el) => ({
          id: el.id,
          name: el.name,
        }));
        this.selectRoles = data.map((el) => ({
          value: el.id,
          label: el.name,
        }));
        console.log(this.roles, 'roles');
      } catch (error) {
        if (error.message === 'Network Error') {
          this.toast.error("Error: Check your network or it's probably a CORS error");
        } else {
          this.toast.error(error.message);
        }
      }
    },
    async getAdmins(pagination) {
      console.log(pagination, 'pagination');
      const limit = pagination.limit || 10;
      const offset = pagination.offset || 0;
      const search = this.searchValue || '';
      this.adminPagination = {
        limit,
        offset,
      };
      try {
        this.loading = true;
        const {
          data: { data },
        } = await API.get(
          `employees?offset=${offset}&limit=${limit}&group=INTERNAL_DASHBOARD&order=desc&sort=id&search=${search}`,
        );
        this.admins = data.map((el) => ({
          ...el,
          name: el.profile.name,
          email: el.profile.email,
          role: el.id || '',
          phone_number: el.profile.phone_number,
          address: el.profile.address,
          is_active: el.is_active,
        }));
        data.forEach((el) => {
          this.is_active[`${el.id}`] = el.is_active;
        });
      } catch (error) {
        if (error.message === 'Network Error') {
          this.toast.error("Error: Check your network or it's probably a CORS error");
        } else {
          this.toast.error(error.message);
        }
      }
      this.loading = false;
    },
    async deleteUser() {
      try {
        this.deleteLoading = true;
        await API.delete(`employees/${this.userId}`);

        this.getAdmins(this.adminPagination);
        this.deleteConfirmation = false;
        this.toast.success('User successfully deleted');
      } catch (error) {
        this.toast.error(error.message);
      }
      this.deleteLoading = false;
    },
    openConfirmation(userId) {
      this.deleteConfirmation = true;
      this.$store.commit('SET_USER_ID', userId);
    },
    handleClose() {
      if (this.addModal) {
        this.addModal = false;
      } else {
        this.editModal = false;
      }
    },
    generateRoleName(row) {
      const filtered = this.roles.filter((el) => el.id === row.role_id);
      return filtered.length ? filtered[0].name : '';
    },
    handleDetail(row, params) {
      if (params === 'edit') {
        this.editModal = true;
      } else {
        this.detailModal = true;
      }
      row.role = this.generateRoleName(row);
      this.$store.commit('SET_ADMIN_DETAIL', row);
    },
    async handleActiveAdmin(row) {
      this.is_active[`${row.id}`] = !this.is_active[`${row.id}`];
      row.is_active = this.is_active[`${row.id}`] && this.is_active[`${row.id}`] === true ? 'TRUE' : 'FALSE';
      const payload = {
        role_id: row.role_id,
        name: row.profile.name,
        is_active: row.is_active,
      };
      try {
        const {
          data: { data },
        } = await API.patch(`/employees/${row.id}`, payload);
        this.toast.success(
          `Success ${data.is_active === true ? 'enable' : 'disable'} admin status !`,
        );
      } catch (error) {
        if (error.message === 'Network Error') {
          this.toast.error("Error: Check your network or it's probably a CORS error");
        } else {
          this.toast.error(error.message);
        }
      }
    },
    async generateFetchData() {
      await this.getAdmins(this.adminPagination);
      await this.getRoles();
    },
  },
  mounted() {
    this.generateFetchData();
    this.$store.state.access.access.permissions.forEach((el) => {
      switch (el.id) {
        case 84:
          this.adminAccess.create = true;
          break;
        case 85:
          this.adminAccess.update = true;
          break;
        case 86:
          this.adminAccess.delete = true;
          break;
        default:
          break;
      }
    });
  },
  computed: {
    userId() {
      return this.$store.state.userId;
    },
  },
};
</script>

<style></style>
