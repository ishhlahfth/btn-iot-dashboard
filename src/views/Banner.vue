<template>
  <help-modal v-model="bannerForm" permanent>
    <banner-form @close="bannerForm = false" @reload="getBanners(bannerPagination)" />
  </help-modal>

  <help-modal v-model="bannerDetail">
    <banner-detail @close="bannerDetail = false" />
  </help-modal>

  <help-modal v-model="deleteConfirmation">
    <confirmation
      title="Delete confirmation"
      message="This action cannot be undone. Are you sure you want to delete this banner permanently?"
      @close="deleteConfirmation = false"
      @cancel="deleteConfirmation = false"
      @confirm="deleteBanner"
    />
  </help-modal>

  <div class="p-4 sm:p-6 grid gap-4 sm:gap-6 auto-rows-max">
    <div class="w-full flex justify-between">
      <p class="text-heading2 font-semibold">Banner</p>
      <div class="grid grid-flow-col gap-4">
        <help-button
          v-if="!reorderMode"
          label="reorder"
          icon="reorder"
          disabled
          @click="reorderMode = true"
        />
        <template v-else>
          <help-button label="cancel" bg-color="flame" @click="reorderMode = false" />
          <help-button label="save" bg-color="mint" />
        </template>
        <help-button label="add" icon="plus" @click="addBanner" />
      </div>
    </div>
    <div class="overflow-hidden">
      <help-table
        path="banners"
        :columns="columns"
        :loading="loading"
        :rows="banners"
        :pagination="bannerPagination"
        @onChangePagination="getBanners($event)"
        @sort="getBanners($event)"
      >
        <template v-slot:body="{ column, row }">
          <div v-if="column === 'banner'" class="grid grid-flow-col gap-2 auto-cols-max">
            <help-thumbnail :src="row.image_url" :height="row.image_url ? '100%' : 72" />
            <div class="flex flex-col justify-between">
              <div>
                <p class="font-semibold">{{ row.title }}</p>
                <a
                  :href="row.hyperlink"
                  target="blank"
                  class="cursor-pointer text-royal hover:underline"
                  >{{ row.hyperlink }}</a
                >
              </div>
              <div class="grid grid-flow-col auto-cols-max gap-2 ">
                <p>{{ row.startDate }}</p>
                <p>-</p>
                <p>{{ row.endDate }}</p>
              </div>
            </div>
          </div>
          <!-- <help-badge
            v-if="column === 'is_active'"
            :label="row.is_active ? 'Active' : 'Inactive'"
            :color="row.is_active ? 'positive' : 'negative'"
          /> -->
          <help-toggle
            v-if="column === 'is_active'"
            v-model="row.is_active"
            @change="toggleBanner($event, row)"
          />
          <div v-if="column === 'actions'" class="grid grid-flow-col gap-1 auto-cols-max">
            <help-button
              bg-color="royal"
              color="white"
              icon="edit"
              icon-only
              @click="editBanner(row)"
            />
            <help-button
              bg-color="flame"
              color="white"
              icon="trash"
              icon-only
              @click="openConfirmation(row.id)"
            />
          </div>
          <p
            v-if="column === 'detail'"
            class="text-royal font-medium cursor-pointer"
            @click="openBannerDetail(row.id)"
          >
            See Detail
          </p>
        </template>
      </help-table>
    </div>
  </div>
</template>

<script>
import BannerDetail from '@/components/modals/BannerDetail.vue';
import BannerForm from '@/components/modals/BannerForm.vue';
import Confirmation from '@/components/modals/Confirmation.vue';
// import HelpBadge from '@/components/atoms/Badge.vue';
import HelpButton from '@/components/atoms/Button.vue';
import HelpModal from '@/components/templates/Modal.vue';
import HelpTable from '@/components/templates/Table.vue';
import HelpToggle from '@/components/atoms/Toggle.vue';
import HelpThumbnail from '@/components/atoms/Thumbnail.vue';
import { useToast } from 'vue-toastification';
import dayjs from 'dayjs';
import API from '../apis';

export default {
  name: 'Banner',
  components: {
    BannerDetail,
    BannerForm,
    Confirmation,
    // HelpBadge,
    HelpButton,
    HelpModal,
    HelpTable,
    HelpToggle,
    HelpThumbnail,
  },
  setup() {
    const toast = useToast();
    return { toast };
  },
  data() {
    return {
      columns: [
        { field: 'banner', label: 'name' },
        { field: 'is_active', label: 'status', align: 'center' },
        { field: 'detail', label: 'detail', align: 'center' },
        { field: 'actions', label: ' ', align: 'center' },
      ],
      banners: [],
      bannerPagination: {
        limit: 10,
        offset: 0,
      },
      loading: false,
      reorderMode: false,
      bannerDetail: false,
      bannerForm: false,
      deleteConfirmation: false,
    };
  },
  computed: {
    bannerId() {
      return this.$store.state.bannerId;
    },
  },
  methods: {
    async getBanners(pagination, withoutLoading = false) {
      const limit = pagination.limit || 10;
      const offset = pagination.offset || 0;

      try {
        if (!withoutLoading) this.loading = true;
        const {
          data: { data },
        } = await API.get(`banners?bannerable=GLOBAL&offset=${offset}&limit=${limit}`);
        this.banners = data.map((el) => ({
          ...el,
          startDate: dayjs(el.start_date).format('ddd, D MMM YYYY'),
          endDate: el.end_date ? dayjs(el.end_date).format('ddd, D MMM YYYY') : 'Forever',
          image_url: '',
        }));
        for (let i = 0; i < data.length; i += 1) {
          const response = await this.$store.dispatch('loadImage', data[i].url);
          this.banners[i].image_url = response;
        }
        this.bannerPagination = {
          limit,
          offset,
        };
      } catch (error) {
        if (error.message === 'Network Error') {
          this.toast.error("Error: Check your network or it's probably a CORS error");
        } else {
          this.toast.error(error.message);
        }
      }
      this.loading = false;
    },
    openBannerDetail(bannerId) {
      this.bannerDetail = true;
      this.$store.commit('SET_BANNER_ID', bannerId);
    },
    addBanner() {
      this.bannerForm = true;
      this.$store.commit('SET_FORM_TYPE', 'ADD');
    },
    editBanner(banner) {
      this.bannerForm = true;
      this.$store.commit('SET_BANNER', banner);
      this.$store.commit('SET_FORM_TYPE', 'EDIT');
    },
    openConfirmation(bannerId) {
      this.deleteConfirmation = true;
      this.$store.commit('SET_BANNER_ID', bannerId);
    },
    async toggleBanner(newValue, detail) {
      const payload = {
        start_date: detail.start_date,
        end_date: detail.end_date,
        title: detail.title,
        hyperlink: detail.hyperlink,
        is_active: newValue,
      };
      try {
        const {
          data: { data },
        } = await API.patch(`banners/${detail.id}`, payload);

        const action = data.is_active ? 'enabled' : 'disabled';
        this.toast.success(`Successfully ${action} ${detail.title}`);
        this.getBanners(this.bannerPagination, true);
      } catch (error) {
        this.toast.error(error.message);
      }
    },
    async deleteBanner() {
      console.log(this.bannerId);
      // try {
      //   const {
      //     data: { data },
      //   } = await API.delete(`banners/${this.bannerId}`);

      //   console.log('ðŸŒ›', data);
      // } catch (error) {
      //   this.toast.error(error.message);
      // }
    },
  },
  mounted() {
    this.getBanners(this.bannerPagination);
  },
};
</script>

<style></style>
