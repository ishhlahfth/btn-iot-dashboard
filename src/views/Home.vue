<template>
  <div class="home bg-black">
    <p class="text-red-700">
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Eaque saepe blanditiis, perspiciatis
      velit quas consequuntur repellendus impedit, distinctio amet sapiente iste. Quae quia
      voluptatum dolorum explicabo nihil dolores asperiores voluptas hic facere nulla modi, delectus
      atque ad eligendi est nisi qui praesentium! Autem fuga quas laborum asperiores odio. Sunt,
      error.
    </p>
  </div>
</template>

<script>
// @ is an alias to /src
// import HelloWorld from '@/components/HelloWorld.vue';

export default {
  name: 'Home',
};
</script>
