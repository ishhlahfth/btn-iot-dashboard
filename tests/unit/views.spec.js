import Home from '@/views/Home.vue';
import Dashboard from '@/views/Dashboard.vue';
import Login from '@/views/Login.vue';
import Role from '@/views/Role.vue';
import Transfer from '@/views/Transfer.vue';
import Profile from '@/views/Profile.vue';
import Payment from '@/views/Payment.vue';
import Order from '@/views/Order.vue';
import Merchant from '@/views/Merchant.vue';
import Invoice from '@/views/Invoice.vue';
import ResetPassword from '@/views/ResetPassword.vue';
import Admin from '@/views/Admin.vue';
import Banner from '@/views/Banner.vue';
import CopyMerchant from '@/views/CopyMerchant.vue';
import InvoiceDownload from '@/views/InvoiceDownload.vue';
import TermsConditions from '@/views/TermsConditions.vue';
import _Components from '@/views/_Components.vue';

describe('Login.vue', () => {
  it('has the expected html structure', () => {
    expect(Login.$el).toMatchSnapshot();
  });

  it('should have "Login" as name', () => {
    expect(Login.name).toBe('Login');
  });
});

describe('Role.vue', () => {
  it('has the expected html structure', () => {
    expect(Role.$el).toMatchSnapshot();
  });

  it('should have "Role" as name', () => {
    expect(Role.name).toBe('Role');
  });
});

describe('Transfer.vue', () => {
  it('has the expected html structure', () => {
    expect(Transfer.$el).toMatchSnapshot();
  });

  it('should have "Transfer" as name', () => {
    expect(Transfer.name).toBe('Transfer');
  });
});

describe('Profile.vue', () => {
  it('has the expected html structure', () => {
    expect(Profile.$el).toMatchSnapshot();
  });

  it('should have "Profile" as name', () => {
    expect(Profile.name).toBe('Profile');
  });
});

describe('Payment.vue', () => {
  it('has the expected html structure', () => {
    expect(Payment.$el).toMatchSnapshot();
  });

  it('should have "Payment" as name', () => {
    expect(Payment.name).toBe('Payment');
  });
});

describe('Order.vue', () => {
  it('has the expected html structure', () => {
    expect(Order.$el).toMatchSnapshot();
  });

  it('should have "Order" as name', () => {
    expect(Order.name).toBe('Order');
  });
});

describe('Merchant.vue', () => {
  it('has the expected html structure', () => {
    expect(Merchant.$el).toMatchSnapshot();
  });

  it('should have "Merchant" as name', () => {
    expect(Merchant.name).toBe('Merchant');
  });
});

describe('Invoice.vue', () => {
  it('has the expected html structure', () => {
    expect(Invoice.$el).toMatchSnapshot();
  });

  it('should have "Invoice" as name', () => {
    expect(Invoice.name).toBe('Invoice');
  });
});

describe('ResetPassword.vue', () => {
  it('has the expected html structure', () => {
    expect(ResetPassword.$el).toMatchSnapshot();
  });

  it('should have "ResetPassword" as name', () => {
    expect(ResetPassword.name).toBe('ResetPassword');
  });
});

describe('Home.vue', () => {
  it('has the expected html structure', () => {
    expect(Home.$el).toMatchSnapshot();
  });

  it('should have "Home" as name', () => {
    expect(Home.name).toBe('Home');
  });
});

describe('Dashboard.vue', () => {
  it('has the expected html structure', () => {
    expect(Dashboard.$el).toMatchSnapshot();
  });

  it('should have "Dashboard" as name', () => {
    expect(Dashboard.name).toBe('Dashboard');
  });
});

describe('Admin.vue', () => {
  it('has the expected html structure', () => {
    expect(Admin.$el).toMatchSnapshot();
  });

  it('should have "Admin" as name', () => {
    expect(Admin.name).toBe('Admin');
  });
});

describe('Banner.vue', () => {
  it('has the expected html structure', () => {
    expect(Banner.$el).toMatchSnapshot();
  });

  it('should have "Banner" as name', () => {
    expect(Banner.name).toBe('Banner');
  });
});

describe('CopyMerchant.vue', () => {
  it('has the expected html structure', () => {
    expect(CopyMerchant.$el).toMatchSnapshot();
  });

  it('should have "CopyMerchant" as name', () => {
    expect(CopyMerchant.name).toBe('CopyMerchant');
  });
});

describe('InvoiceDownload.vue', () => {
  it('has the expected html structure', () => {
    expect(InvoiceDownload.$el).toMatchSnapshot();
  });

  it('should have "InvoiceDownload" as name', () => {
    expect(InvoiceDownload.name).toBe('InvoiceDownload');
  });
});

describe('TermsConditions.vue', () => {
  it('has the expected html structure', () => {
    expect(TermsConditions.$el).toMatchSnapshot();
  });

  it('should have "TermsConditions" as name', () => {
    expect(TermsConditions.name).toBe('TermsConditions');
  });
});

describe('_Components.vue', () => {
  it('has the expected html structure', () => {
    expect(_Components.$el).toMatchSnapshot();
  });

  it('should have "_Components" as name', () => {
    expect(_Components.name).toBe('Components');
  });
});

