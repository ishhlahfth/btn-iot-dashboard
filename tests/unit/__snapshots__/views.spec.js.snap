// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_Components.vue has the expected html structure 1`] = `undefined`;

exports[`Admin.vue has the expected html structure 1`] = `undefined`;

exports[`Banner.vue has the expected html structure 1`] = `undefined`;

exports[`CopyMerchant.vue has the expected html structure 1`] = `undefined`;

exports[`Dashboard.vue has the expected html structure 1`] = `undefined`;

exports[`Home.vue has the expected html structure 1`] = `undefined`;

exports[`Invoice.vue has the expected html structure 1`] = `undefined`;

exports[`InvoiceDownload.vue has the expected html structure 1`] = `undefined`;

exports[`Login.vue has the expected html structure 1`] = `undefined`;

exports[`Merchant.vue has the expected html structure 1`] = `undefined`;

exports[`Order.vue has the expected html structure 1`] = `undefined`;

exports[`Payment.vue has the expected html structure 1`] = `undefined`;

exports[`Profile.vue has the expected html structure 1`] = `undefined`;

exports[`ResetPassword.vue has the expected html structure 1`] = `undefined`;

exports[`Role.vue has the expected html structure 1`] = `undefined`;

exports[`TermsConditions.vue has the expected html structure 1`] = `undefined`;

exports[`Transfer.vue has the expected html structure 1`] = `undefined`;
